src/Schema/Directives/LikeDirective.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Nuwave\Lighthouse\Schema\Directives;
+
+use Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective;
+
+class LikeDirective extends BaseDirective implements ArgBuilderDirective
+{
+    const ESCAPE_CHAR = "\\";
+
+    public static function definition(): string
+    {
+        return /** @lang GraphQL */ <<<'GRAPHQL'
+enum LikePercentageLocation {
+    START
+    END
+    BOTH
+}
+
+"""
+Use the client given value to add a `LIKE` conditional to a database query.
+"""
+directive @like(
+  """
+  Specify the database column to compare.
+  Only required if database column has a different name than the attribute in your schema.
+  """
+  key: String
+
+  """
+  Specify the positions of the % in the LIKE comparison. The default is BOTH.
+  """
+  percentage: LikePercentageLocation
+) repeatable on ARGUMENT_DEFINITION | INPUT_FIELD_DEFINITION
+GRAPHQL;
+    }
+
+    protected function escapePercentage(string $value): string
+    {
+        return str_replace(
+            [self::ESCAPE_CHAR, '%', '_'],
+            [self::ESCAPE_CHAR.self::ESCAPE_CHAR, self::ESCAPE_CHAR.'%', self::ESCAPE_CHAR.'_'],
+            $value
+        );
+    }
+
+    /**
+     * Apply a "WHERE LIKE $value" clause.
+     *
+     * @param  \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder  $builder
+     * @return \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder
+     */
+    public function handleBuilder($builder, $value): object
+    {
+        if (!$value) {
```

Comment:
This bails on `''`, which could be a reasonable value for clients to search for when they can control wildcards.

Reply:
handling only null now.